### PR TITLE
Integrate with VIM for day2 operation

### PR DESCRIPTION
**Pull request by ttakenak**
This change is to integrate with VIM:
- Create vim client
- Send strategy create request
- Send strategy apply request
- Send strategy remove request
- Monitor strategy status
- Log reason if strategy fails
- Update strategy request status for each resource

Test Plan:

- Success cases PASS: Fresh install in standalone system
      (SX,DX,STANDARD,STORAGE)
PASS: Fresh install in DC
PASS: Finish successfully to apply day2 config
      in standalone system(SX,DX,STANDARD,STORAGE)
PASS: Finish successfully to apply day2 config
      in DC
PASS: Finish successfully to apply day2 config
      in insync=false node
PASS: Finish successfully when other strategy
      exists but deleted before retry exceeds

- Failure cases PASS: Ansible detects failure for wrong configuration
      (Complete and partial wrong configuration)
PASS: Remove strategy when strategy-build failed
PASS: Remove strategy when applied in locked status node PASS: Retry exceeds to create strategy when another
      strategy exists

### CODE REVIEW

**Review by enzocandotti — APPROVED** [review]
LGTM

File: api/v1/datanetwork_types.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ type DataNetworkStatus struct {
 // attributes of a StarlingX system.  This is a composition of the following
 // StarlingX API endpoints.
 //
-//   https://docs.starlingx.io/api-ref/stx-config/api-ref-sysinv-v1-config.html#data-networks
+//	https://docs.starlingx.io/api-ref/stx-config/api-ref-sysinv-v1-config.html#data-networks
```
Comment (yjian118) [inline api/v1/datanetwork_types.go:112]: I noticed this kind of format change during reviewing the package update in Ritik's recent commit. Is it introduced automatically?

File: controllers/host/host_controller.go [hunk 2]
```diff
@@ -660,6 +660,7 @@ func (r *HostReconciler) ReconcileInitialState(client *gophercloud.ServiceClient
 			}
 
 			logHost.Info("locking host", "opts", opts)
+			logHost.V(2).Info("locking host", "opts", opts)
```
Comment (yjian118) [inline controllers/host/host_controller.go:663]: Looks like it is a redundant v2 level log

File: controllers/host/host_controller.go [hunk 3]
```diff
@@ -960,14 +961,26 @@ func (r *HostReconciler) CompareEnabledAttributes(in *starlingxv1.HostProfileSpe
 // CompareDisabledAttributes determines if two profiles are identical for the
 // purpose of reconciling any attributes that can only be applied when the host
 // is disabled.
-func (r *HostReconciler) CompareDisabledAttributes(in *starlingxv1.HostProfileSpec, other *starlingxv1.HostProfileSpec, namespace, personality string) bool {
+func (r *HostReconciler) CompareDisabledAttributes(in *starlingxv1.HostProfileSpec, other *starlingxv1.HostProfileSpec, namespace, personality string, daytwo bool) bool {
```
Comment (yjian118) [inline controllers/host/host_controller.go:964]: "principal" will be better to understand than "daytwo"

File: controllers/host/host_controller.go [hunk 4]
```diff
@@ -960,14 +961,26 @@ func (r *HostReconciler) CompareEnabledAttributes(in *starlingxv1.HostProfileSpe
...
 	if other == nil {
 		return false
 	}
 
+	// In day2 operation, copy AdministrativeState temporary to ignore difference
+	// Put original value back after DeepEqual
+	adminState := other.ProfileBaseAttributes.AdministrativeState
+	if daytwo {
```
Comment (yjian118) [inline controllers/host/host_controller.go:972]: Don't forget to change it if you accept to use principal than daytwo

File: api/v1/datanetwork_types.go (hunk 1 above)
Comment (ttakenak) [reply api/v1/datanetwork_types.go:112]: I'm using MS code and it automatically formatted all codes like this every time I open and save (even the code which I did not modify). I could put it back the format, so please let me know, thank you

Comment (yjian118) [reply api/v1/datanetwork_types.go:112]: Is it changed recently? If that's a one-time deal and will not be changed the next time, I'll prefer to leave it changed as multiple people hit it. Changing a lot of formats is difficult to review.

**Review by yjian118 — CHANGES_REQUESTED** [review]
Please remove the redundant log and change the "daytwo" variable name if you agree, others LGTM

**Comment by hugonicodemos** [comment]
Should we create unit tests for this?

**Review by yjian118 — APPROVED** [review]

**Comment by yjian118** [comment]
@hugonicodemos @ttakenak  We do need the unit tests for this commit. I'm okay to merge it as-is, and @ttakenak please add unit tests in the next commit

**Pull request merged**